include/fg/defines.h

```diff
@@ -47,10 +47,19 @@
  */
 FGAPI GLEWContext* glewGetContext();
 
-namespace fg
-{
-
-enum ErrorCode {
+typedef void* fg_window;
+typedef void* fg_font;
+typedef void* fg_chart;
+typedef void* fg_image;
+typedef void* fg_histogram;
+typedef void* fg_plot;
+typedef void* fg_surface;
+
+typedef unsigned int    uint;
+typedef unsigned short  ushort;
+typedef unsigned char   uchar;
+
+typedef enum {
     FG_SUCCESS            = 0,              ///< Fuction returned successfully.
```

Comment:
I like the idea of having `FG_ERR_NONE` as opposed to `FG_SUCCESS` (or at least have both) for consistency reasons.